integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanRefundTransactionTest.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.fineract.client.models.PostLoansRequest;
+import org.apache.fineract.client.models.PostLoansResponse;
+import org.apache.fineract.integrationtests.common.BusinessDateHelper;
+import org.apache.fineract.integrationtests.common.ClientHelper;
+import org.apache.fineract.integrationtests.common.LoanRescheduleRequestHelper;
+import org.apache.fineract.integrationtests.common.Utils;
+import org.apache.fineract.integrationtests.common.accounting.AccountHelper;
+import org.apache.fineract.integrationtests.common.loans.LoanProductTestBuilder;
+import org.apache.fineract.integrationtests.common.loans.LoanTransactionHelper;
+import org.apache.fineract.portfolio.common.domain.DaysInMonthType;
+import org.apache.fineract.portfolio.common.domain.DaysInYearType;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoanRefundTransactionTest extends BaseLoanIntegrationTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LoanRefundTransactionTest.class);
+    private static final String DATETIME_PATTERN = "dd MMMM yyyy";
+    private static ResponseSpecification responseSpec;
+    private static RequestSpecification requestSpec;
+    private static BusinessDateHelper businessDateHelper;
+    private static LoanTransactionHelper loanTransactionHelper;
+    private static AccountHelper accountHelper;
+    private static LoanRescheduleRequestHelper loanRescheduleRequestHelper;
+
+    @BeforeAll
+    public static void setup() {
+        Utils.initializeRESTAssured();
+        requestSpec = new RequestSpecBuilder().setContentType(ContentType.JSON).build();
+        requestSpec.header("Authorization", "Basic " + Utils.loginIntoServerAndGetBase64EncodedAuthenticationKey());
+        requestSpec.header("Fineract-Platform-TenantId", "default");
+        responseSpec = new ResponseSpecBuilder().expectStatusCode(200).build();
+        loanTransactionHelper = new LoanTransactionHelper(requestSpec, responseSpec);
+        businessDateHelper = new BusinessDateHelper();
+        accountHelper = new AccountHelper(requestSpec, responseSpec);
+        ClientHelper clientHelper = new ClientHelper(requestSpec, responseSpec);
+        loanRescheduleRequestHelper = new LoanRescheduleRequestHelper(requestSpec, responseSpec);
+    }
+
+    // UC1: Generate a totalInterestRefund using Advanced payment allocation with Interest Refund options and Apply
+    // Interest Refund transaction
+    // 1. Create a Loan product with Adv. Pment. Alloc. and with Accrual and Interest Refund
+    // 2. Submit, Approve and Disburse a Loan account
+    // 3. Apply the MERCHANT_ISSUED_REFUND transaction to validate totalInterestRefund different than Zero
+    // 4- Apply the INTEREST_REFUND transaction to validate the Journal Entries generated
+    @Test
+    public void uc1() {
+        final String operationDate = "1 January 2024";
```

Comment:
This test is not really testing whether the correct journal entries were used when the interest refund had the following portions for example:
- principal 5
- fee 4
- penalty 3
- interest 2
- overpayment 1

Also there is no use case to cover charged off situations

Please kindly add further test cases to cover the accounting requirements!